src/SAWScript/Crucible/MIR/Override.hs

```diff
@@ -117,6 +136,400 @@ assignTerm sc cc md prepost var val =
+  PC.compareF ctx1 ctx2 <<>>
+  PC.compareF idx1 idx2 <<>>
+  cmpPathConcretely sym p1 p2 <<>>
+  PC.EQF
+cmpPathConcretely _ (Mir.Field_RefPath _ _ _) _ = PC.LTF
+cmpPathConcretely _ _ (Mir.Field_RefPath _ _ _) = PC.GTF
+
+-- Variant_RefPath
+cmpPathConcretely sym (Mir.Variant_RefPath discrTp1 ctx1 p1 idx1) (Mir.Variant_RefPath discrTp2 ctx2 p2 idx2) =
+  PC.compareF discrTp1 discrTp2 <<>>
+  PC.compareF ctx1 ctx2 <<>>
+  PC.compareF idx1 idx2 <<>>
+  cmpPathConcretely sym p1 p2 <<>>
+  PC.EQF
+cmpPathConcretely _ (Mir.Variant_RefPath _ _ _ _) _ = PC.LTF
+cmpPathConcretely _ _ (Mir.Variant_RefPath _ _ _ _) = PC.GTF
+
+-- Index_RefPath
+cmpPathConcretely sym (Mir.Index_RefPath tpr1 p1 i1) (Mir.Index_RefPath tpr2 p2 i2) =
+  PC.compareF tpr1 tpr2 <<>>
+  cmpPathConcretely sym p1 p2 <<>>
+  cmpSymBVConcretely sym i1 i2 <<>>
+  PC.EQF
+cmpPathConcretely _ (Mir.Index_RefPath _ _ _) _ = PC.LTF
+cmpPathConcretely _ _ (Mir.Index_RefPath _ _ _) = PC.GTF
+
+-- Just_RefPath
+cmpPathConcretely sym (Mir.Just_RefPath tpr1 p1) (Mir.Just_RefPath tpr2 p2) =
+  PC.compareF tpr1 tpr2 <<>>
+  cmpPathConcretely sym p1 p2 <<>>
+  PC.EQF
+cmpPathConcretely _ (Mir.Just_RefPath _ _) _ = PC.LTF
+cmpPathConcretely _ _ (Mir.Just_RefPath _ _) = PC.GTF
+
+-- VectorAsMirVector_RefPath
+cmpPathConcretely sym (Mir.VectorAsMirVector_RefPath tpr1 p1) (Mir.VectorAsMirVector_RefPath tpr2 p2) =
+  PC.compareF tpr1 tpr2 <<>>
+  cmpPathConcretely sym p1 p2 <<>>
+  PC.EQF
+cmpPathConcretely _ (Mir.VectorAsMirVector_RefPath _ _) _ = PC.LTF
+cmpPathConcretely _ _ (Mir.VectorAsMirVector_RefPath _ _) = PC.GTF
+
+-- ArrayAsMirVector_RefPath
+cmpPathConcretely sym (Mir.ArrayAsMirVector_RefPath tpr1 p1) (Mir.ArrayAsMirVector_RefPath tpr2 p2) =
+  PC.compareF tpr1 tpr2 <<>>
+  cmpPathConcretely sym p1 p2 <<>>
+  PC.EQF
+
+-- | Compare two 'W4.SymBV' values that are known to be concrete. If they are
+-- not concrete, this function will panic.
+-- See @Note [MIR compositional verification and mutable allocations]@.
+cmpSymBVConcretely ::
+  Crucible.IsSymInterface sym =>
+  proxy sym ->
+  W4.SymBV sym w ->
+  W4.SymBV sym w ->
+  PC.OrderingF w w
+cmpSymBVConcretely _ symBV1 symBV2
+  | Just bv1 <- W4.asBV symBV1
+  , Just bv2 <- W4.asBV symBV2
+  = PC.fromOrdering $ compare bv1 bv2
+  | otherwise
+  = panic "cmpSymBVConcretely"
+          [ "SymBV not concrete"
+          , show $ W4.printSymExpr symBV1
+          , show $ W4.printSymExpr symBV2
+          ]
+
+-- | An infix version of 'PC.joinOrderingF' that is right associative, allowing
+-- it to be chained together more easily. See
+-- <https://github.com/GaloisInc/parameterized-utils/issues/157> for further
+-- motivation.
+infixr 6 <<>>
+(<<>>) ::
+  forall j k (a :: j) (b :: j) (c :: k) (d :: k).
+  PC.OrderingF a b ->
+  (a ~ b => PC.OrderingF c d) ->
+  PC.OrderingF c d
+(<<>>) = PC.joinOrderingF
+
+{-
+Note [MIR compositional verification and mutable allocations]
```

Comment:
A Note which describes the `checkMutableAllocPostconds` function at a high level.